### PR TITLE
Add post editing

### PR DESCRIPTION
**Pull request by ksnavely**
Here I add the ability to edit existing posts. Once logged in, an 'Edit' option is available per post on the front-page. Navigating there will display a page very similar to the create-post path, with a preview option available.

I've tested this manually in staging.

### CODE REVIEW

File: website/accounts.py [hunk 1]
```diff
@@ -11,10 +11,6 @@
 dispensed with a lot of the validation and niceties one might expect
 in production. Use at your own risk.
 """
-import hashlib
```
Comment (ksnavely) [inline website/accounts.py:14]: I also performed a pyflakes pass which eliminated a few unused imports and variables.

File: website/server/templates/index.html [hunk 2]
```diff
@@ -26,16 +26,17 @@ <h1 class="blog-title">kdevops</h1>
...
-					{{ post.text|safe }}
+                  <div class="blog-post">
+                    <div class="blog-post-title"><h2>{{ post.title }}</h2></div>
+                    {% if current_user.is_authenticated %}<div class="blog-post-edit"><a href="{{ url_for('update_post', post_id=post._id) }}">Edit</a></div>{% endif %}
+                    <p class="blog-post-meta">{{ post.date }} by <a href="#">{{ post.author }}</a></p>
+                    {{ post.text|safe }}
                   <p class="blog-post-meta">Tags: {% for t in post.tags %} {{ t }} {% endfor %}</p>
 
-				  </div><!-- /.blog-post --> 
+                  </div><!-- /.blog-post -->
```
Comment (ksnavely) [inline website/server/templates/index.html:36]: change tabs for spaces

what kind of jerk uses tabs anyways

**Comment by ksnavely** [comment]
The reusability of templates will be improved in the future.

File: website/server/templates/index.html (hunk 2 above)
Comment (evansde77) [inline website/server/templates/index.html:36]: The worst kind

File: website/server/views/frontpage.py [hunk 3]
```diff
@@ -44,11 +60,42 @@ def create_blog_entry_submit():
...
+    """
+    Updates an existing blog post.
+    """
+    if not request.form:
+        return (
+            "update_blog_entry: Blog entry form data is required.",
+            404)
+
+
+    required_fields = ["title", "tags", "text", "author", "date"]
```
Comment (evansde77) [inline website/server/views/frontpage.py]: Module level constant?

**Comment by evansde77** [comment]
LGTM although I'm leery about defining the main app in an init.py (one of my biggest gripes about Wilson)

**Comment by ksnavely** [comment]
Thanks for the comments Dave!

**Pull request merged**